---
layout: splash
author_profile: false
permalink: /publication/
title: "Publication"
---
<!-- link rel="stylesheet" href="https://czheluo.github.io/assets/css/styles.css" -->
<h3 align="left">Published:*Co-ﬁrst Author Publications; #contributed equally to this work.</h3>

<li>#Tao Li, #Meng Luo, Dadong Zhang, Di Wu, Lei Li, Guihua Bai. Effective marker alleles associated with type 2 resistance of wheat to Fusarium head blight infection in fields. Breeding Science, 2016, 66(3):350-357.</li>
<li>#Li Tao, #Meng Luo, Qian Dan, Dong JingJing, Gu ShiLiang. Evaluation of Phenotypic Variation of EMS-induced Mutants of Chinese Wheat Landraces Huangfangzhu and Haiyanzhong Resistant to Fusarium Head Blight. Journal of Plant Genetic Resources,2016, doi:10.13430/j.cnki.jpgr.2016.06.018. (In chinese). Core Journals!</li>
<li>*Qian Dan, #Meng Luo, Dong Jingjing, Li Changcheng, Li Lei, Li Tao. Evaluation of Agronomic Traits and Resistance to Fusarium Head Blight of Ning 7840 and Its Mutants. Journal of Triticeae Crops, 2016, doi:10.7606/j.issn.1009-1041.2016.02.16. (In chinese). Core Journals!</li>
<li>#Gong Xuan, #Meng Luo, Xiao Tianjing, Cao Jing, Li Lei, Li Tao. Variations in agronomic traits and resistance to Fusarium head blight in wheat mutants. Journal of Yangzhou University (Agricultural and Life Science Edition), 2017, doi: 10.16872/j.cnki.1671-4652.2017.04.011. (In chinese). Core Journals!</li>
<li>SHI Xuan, LI Lei, ZHENG Tong, LUO Meng, LI Tao. Differential Responses of Wheat Mutants with Hypersensitive Reaction-Like Trait to Nitrogen Dosage and Powdery Mildew Infection. Journal of Triticeae Crops. 2018,(38)04,410-417 DOI:10.7606/j.issn.1009-1041.2018.</li>
<h3 align="left">Preprint</h3>

<li> Luo, M. & Gu, S. <a href="https://www.biorxiv.org/content/10.1101/2020.10.16.336180v1">A new approach of dissecting genetic effects for complex traits. bioRxiv, 2020.10.16.336180 (2020).</a></li>
<li> Meng Luo, Shiliang Gu. Iterative Screen Regression Models for Genetic Mapping Studies of Epistasis of Quantitative Traits. 2018, (Current preprint (BioRxiv)). My blog.</li>
<li> Meng Luo, Shiliang Gu. Genetic Prediction of Complex Traits with Iterative Screen Regression Models. 2018, (Current preprint (BioRxiv)). My blog.</li>
<li> Luo, Meng; Gu, Shiliang (2020): <a href="https://doi.org/10.36227/techrxiv.13139042.v2"> Solve traveling salesman problem using EMF-CE algorithm. TechRxiv. Preprint.</a></li>
<li> #Meng Luo, #ZHANG Ming-Yan, PENG Yong-Xin, GUO Wen-Shan, ZHU Xin-Gai, LI Chun-Yan, BAI Gui-Hua, LI Tao and FENG Chao-Nian. Genome-Wide Association Analysis between SNP Markers and Zinc Content in Wheat Grains. Acta Agronomica Sinica, (in Chinese and under review). Core Journals!</li>

<h3 align="left">Team Work project in MajorBio</h3>

<link type="text/css" rel="stylesheet" href="https://czheluo.github.io/assets/bootstrap-3.3.7/css/timelines.css" >
<link type="text/css" rel="stylesheet" href="https://czheluo.github.io/assets/bootstrap-3.3.7/css/bootstrap.min.css" >

<section id="publications">
  <div class="container" style="padding: 0px; margin: 0px; width: 100%">
    <div class="row">
      <div class="col-lg-12 text-center">
        <h2 class="section-heading">NEWS &amp; PUBLICATIONS</h2>
        <h3 class="section-subheading text-muted"></h3>
      </div>
    </div>
    <div class="row">
      <div class="col-lg-12">
        <ul class="timeline">
            <li>
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/nature_cell_biology_twitter_logo.jpg" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Robust gene expression programs underlie recurrent cell states and phenotype switching in melanoma</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Melanoma cells can switch between a melanocytic and a mesenchymal-like state. Scattered evidence indicates that additional intermediate state(s) may exist. Here, to search for such states and decipher their underlying gene regulatory network (GRN), we studied 10 melanoma cultures using single-cell RNA sequencing (RNA-seq) as well as 26 additional cultures using bulk RNA-seq.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/10.1101/715995v2" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.nature.com/articles/s41556-020-0547-3" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
            
            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/csh_genome_research_twitter_logo.jpg" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Cross-species analysis of enhancer logic using deep learning</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Genomic enhancers form the central nodes of gene regulatory networks by harbouring combinations of transcription factor binding sites. In order to unravel the enhancer logic of the two most common melanoma cell states, namely the melanocytic and mesenchymal-like state, we combined comparative epigenomics with machine learning. By profiling chromatin accessibility using ATAC-seq on a cohort of 27 melanoma cell lines across six different species, we demonstrate the conservation of the two main melanoma states and their underlying master regulators.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/10.1101/2019.12.21.885715v1" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://genome.cshlp.org/content/early/2020/07/30/gr.260844.120.abstract" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
            
            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/msb_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Identification of genomic enhancers through spatial integration of single‐cell transcriptomics and epigenomics</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted"></p><p>Single‐cell technologies allow measuring chromatin accessibility and gene expression in each cell, but jointly utilizing both layers to map bona fide gene regulatory networks and enhancers remains challenging. Here, we generate independent single‐cell RNA ‐seq and single‐cell ATAC ‐seq atlases of the Drosophila eye‐antennal disc and spatially integrate the data into a virtual latent space that mimics the organization of the 2D tissue using ScoMAP (Single‐Cell Omics Mapping into spatial Axes using Pseudotime ordering).</p>

<p><img src="img/publications/msb_cover_scomap.jpeg" width="300" hspace="200" vspace="10"></p>
<p></p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/10.1101/2019.12.19.882381v1.full" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.embopress.org/doi/10.15252/msb.20209438" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
            
            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/bioarxiv_logo_preprint_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Prioritization of enhancer mutations by combining allele-specific chromatin accessibility with deep learning</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Prioritization of non-coding genome variation benefits from explainable AI to predict and interpret the impact of a mutation on gene regulation. Here we apply a specialized deep learning model to phased melanoma genomes and identify functional enhancer mutations with allelic imbalance of chromatin accessibility and gene expression.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/10.1101/2019.12.21.885806v1" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
            
            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/bioarxiv_logo_preprint_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Identification of genomic enhancers through spatial integration of single-cell transcriptomics and epigenomics</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Single-cell technologies allow measuring chromatin accessibility and gene expression in each cell, but jointly utilizing both layers to map bona fide gene regulatory networks and enhancers remains challenging. Here, we generate independent single-cell RNA-seq and single-cell ATAC-seq atlases of the Drosophila eye-antennal disc and spatially integrate the data using a virtual latent space that mimics the organization of the 2D tissue.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/10.1101/2019.12.19.882381v1" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
            
            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/bioarxiv_logo_preprint_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Single-cell gene regulatory network analysis reveals new melanoma cell states and transition trajectories during phenotype switching</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Melanoma is notorious for its cellular heterogeneity. Similarly to EMT, melanoma cells with a melanocytic phenotype can switch to a mesenchymal-like phenotype. However, scattered emerging evidence indicates that additional, intermediate state(s) may exist. In order to search for such new melanoma states and decipher their underlying gene regulatory network (GRN), we extensively studied ten patient-derived melanoma cultures by single-cell RNA-seq of &gt;39,000 cells.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/10.1101/715995v1" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
            
            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/stein_aerts_nature_interview.jpg" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Author File in Nature Methods features Stein Aerts</h4>
                  <div>Nature Methods 16, 355 (2019)</div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">“It’s still me,” says Stein Aerts about his diverse science chapters to date. […] Click on link below to read more.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                    
                    <a class="btn btn-primary btn-outline btn-link" href="https://www.nature.com/articles/s41592-019-0397-8" target="_blank" rel="noopener">Read more</a>
                    
                    
                </div>
              </div>
            </li>
            
          
          
          
          
          
          
          
            
            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/nature_methods_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">cisTopic: cis-regulatory topic modeling on single-cell ATAC-seq data.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Single-cell epigenomics provides new opportunities to decipher genomic regulatory programs from heterogeneous samples and dynamic processes. We present a probabilistic framework called cisTopic, to simultaneously discover “cis-regulatory topics” and stable cell states from sparse single-cell epigenomics data.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/node/112231.article-info" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/30962623" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
 
            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/cell_twitter_logo.jpg" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">A Single-Cell Transcriptome Atlas of the Aging Drosophila Brain.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">A single-cell atlas of the adult fly brain during aging:</p>

<ul>
<li>Network inference reveals regulatory states related to oxidative phosphorylation</li>
<li>Cell identity is retained during aging despite exponential decline of gene expression</li>
<li><a href="http://scope.aertslab.org">SCope</a>: An online tool to explore and compare single-cell datasets across species</li>
</ul><p></p>
                </div>
                <div style="margin-top: 5px;">
                    
                      
                        <a class="btn btn-primary btn-outline btn-link" href="https://www.biorxiv.org/content/early/2017/12/21/237420" target="_blank" rel="noopener">Preprint</a>
                      
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/29909982" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>

            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/nature_genetics_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">The transcription factor Grainy head primes epithelial enhancers for spatiotemporal activation by displacing nucleosomes.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Using ATAC-seq across a panel of Drosophila inbred strains, we found that SNPs affecting binding sites of the TF Grainy head (Grh) causally determine the accessibility of epithelial enhancers.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/29867222" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>

            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/data_software_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Data/Software Update</h4>
                  <div>Data &amp; Software - Mar. 30th 2018</div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted"></p><ul>
<li><a href="https://scenic.aertslab.org/">SCENIC</a> is now available for <strong>Fly</strong>.</li>
<li><a href="https://arboreto.readthedocs.io/en/latest/">Arboreto</a> makes GENIE3 highly parallelizable and comes with a novel and fast GRN inference algorithm as an alternative for GENIE3 for very large datasets.</li>
<li><a href="http://pyscenic.readthedocs.io/en/latest/">pySCENIC</a> is a lightning-fast python implementation of the SCENIC pipeline (Single-CEll regulatory Network Inference and Clustering).</li>
</ul><p></p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://scenic.aertslab.org/" target="_blank" rel="noopener">Website</a>
                    
                </div>
              </div>
            </li>

            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/fly_cell_atlas_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Fly Cell Atlas</h4>
                  <div>Gasthuisberg campus, University of Leuven (Belgium), Dec. 8th 2017</div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">The Fly Cell Atlas will bring together Drosophila researchers interested in single-cell genomics, transcriptomics, and epigenomics, to build comprehensive cell atlases during different developmental stages and disease models.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="http://flycellatlas.org/" target="_blank" rel="noopener">Website</a>
                    
                </div>
              </div>
            </li>

            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/nature_methods_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">SCENIC: single-cell regulatory network inference and clustering.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">We present <a href="https://scenic.aertslab.org/">SCENIC</a>, a computational method for simultaneous gene regulatory network reconstruction and cell-state identification from single-cell RNA-seq data.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/28991892" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>

            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/developmental_cell_twitter_logo.png" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Hippo Reprograms the Transcriptional Response to Ras Signaling.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Here we show that the Hippo pathway is critical for this decision. Loss of Hippo switches Ras activation from promoting cellular differentiation to aggressive cellular proliferation.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/28950103" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
 
            <li>
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/csh_genome_research_twitter_logo.jpg" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Multiplex enhancer-reporter assays uncover unsophisticated TP53 enhancer logic.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Using two complementary techniques of multiplex enhancer-reporter assays, we discovered that functional enhancers could be discriminated from nonfunctional binding events by the occurrence of a single TP53 canonical motif.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/27197205" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
            <li class="timeline-inverted">
              
              
              <div class="timeline-image">
                
                <img class="img-circle img-responsive" src="https://aertslab.org/img/publications/nature_communications_twitter_logo.jpeg" alt="">
              </div>
              <div class="timeline-panel">
                <div class="timeline-heading">
                  <h4 class="subheading">Decoding the regulatory landscape of melanoma reveals TEADS as regulators of the invasive cell state.</h4>
                  <div></div>
                </div>
                <div class="timeline-body">
                  <p class="text-muted">Using regulatory landscapes and in silico analysis, we show that transcriptional reprogramming underlies the distinct cellular states present in melanoma. Furthermore, it reveals an essential role for the TEADs, linking it to clinically relevant mechanisms such as invasion and resistance.</p>
                </div>
                <div style="margin-top: 5px;">
                    
                    
                      <a class="btn btn-primary btn-outline btn-link" href="https://www.ncbi.nlm.nih.gov/pubmed/25865119" target="_blank" rel="noopener">Paper</a>
                    
                    
                    
                </div>
              </div>
            </li>
          
          <li class="timeline-inverted">
            <div class="timeline-image">
              <a href="#publication-timeline" data-toggle="modal" style="color: white">
                <div class="tl-end-point">See all <br> of our<br>publications!</div>
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>



<!-- div class="item">
  <div class="image">
    <div>
      <img src="http://lorempixel.com/400/400/people/1/" />
    </div>
  </div>
  <div class="details">
    <div>
      <h1>Phasellus pharetra</h1>
      <p>Proin condimentum tempus ultrices. Suspendisse vestibulum suscipit erat, ac efficitur lorem. Nullam non ex vel turpis imperdiet maximus sit amet nec odio. Donec mauris nisl, vestibulum id efficitur at, convallis id dui. Sed enim nisl, ultrices vitae sodales eu, vestibulum a mi. Morbi consectetur pulvinar sagittis. Phasellus pharetra diam id leo gravida pharetra. In rutrum est gravida, maximus mi ac, mattis metus. Ut at tempus sem. Vivamus condimentum erat eget aliquet dignissim. </p>
    </div>
  </div>
</div>

<div class="item">
  <div class="image">
    <div>
      <img src="http://lorempixel.com/400/400/people/3/" />
    </div>
  </div>
  <div class="details">
    <div>
      <h1>Proin consectetur</h1>
      <p>Quisque vestibulum iaculis felis eget varius. Morbi bibendum lectus felis, sed vehicula purus pellentesque eu. Maecenas sit amet libero vel nunc porta tincidunt fringilla nec mauris. Phasellus pharetra pharetra massa. Ut maximus purus eget libero dapibus, sit amet ultricies eros auctor. Nulla tristique dui nec libero rhoncus, et dictum odio lobortis. Proin consectetur nec sapien hendrerit lobortis. Proin mi odio, mollis ac ullamcorper in, pellentesque non nulla. In pretium purus at posuere efficitur. Donec eget cursus quam, in mollis libero. Mauris tempor volutpat sodales. Nulla in odio nec metus elementum ultricies sit amet vel dolor. Nunc non ligula at libero eleifend consectetur nec eu felis. Sed tristique vel ante eu pulvinar. Pellentesque elementum congue iaculis. Integer lorem sapien, vestibulum sed semper sed, dictum nec purus. Aenean porttitor interdum iaculis. </p>
    </div>
  </div>
</div>

<div class="item">
  <div class="image">
    <div>
      <img src="http://lorempixel.com/400/400/people/8/" />
    </div>
  </div>
  <div class="details">
    <div>
      <h1>Duis porta urna</h1>
      <p>In nulla mi, ultrices et ullamcorper molestie, rhoncus at purus. Nulla et metus elementum, molestie lectus eget, pretium sapien. Duis dapibus libero id vehicula sodales. Curabitur augue tellus, lacinia ac turpis sit amet, facilisis finibus mauris. Mauris laoreet neque nisi, id hendrerit diam cursus ut. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Duis porta urna ligula, nec consequat libero congue quis. Nulla convallis enim sed dolor viverra, eget dapibus lorem ultricies. Nullam et neque vitae nulla faucibus sollicitudin eget sit amet justo.</p>
    </div>
  </div>
</div -->
